frontend/pages/register.vue

```diff
@@ -0,0 +1,261 @@
+<script setup lang="ts">
+import { useField, useForm } from 'vee-validate'
+import { REGEXP_VALID_EMAIL } from '~/utils/regexp'
+import { Target } from '~/types/links'
+import { tOf } from '~/utils/translation'
+import { API_PATH } from '~/types/customFetch'
+
+const { t: $t } = useI18n()
+const { fetch } = useCustomFetch()
+const toast = useBcToast()
+
+useBcSeo('login_and_register.title_register')
+
+const { handleSubmit, errors } = useForm()
+const { value: email } = useField<string>('email', validateAddress)
+const { value: password1 } = useField<string>('password1', validatePassword1)
+const { value: password2 } = useField<string>('password2', validatePassword2)
+const { value: agreement } = useField<boolean>('agreement', validateAgreement)
+
+function validateAddress (value: string) : true|string {
+  if (!value) {
+    return $t('login_and_register.no_email')
+  }
+  if (value.length > 100 || !REGEXP_VALID_EMAIL.test(value)) {
+    return $t('login_and_register.invalid_email')
+  }
+  return true
+}
```

Comment:
**Suggestion:** This duplicates code from here
https://github.com/gobitfly/beaconchain/blob/eb45e79d507af5db9ea0132f0db54d11cad288e0/frontend/pages/login.vue#L17-L25

Please move it to a shared location.

The same applies with `validatePassword1` and this
https://github.com/gobitfly/beaconchain/blob/eb45e79d507af5db9ea0132f0db54d11cad288e0/frontend/pages/login.vue#L27-L35

Both have an exception: In `login.vue` the maximum length is not checked. This could be added for consistency reasons. Or maybe add a parameter and if it is undefined the maximum length is not checked in the function.

`validatePassword2` is very similar and I suggest to just add a parameter (something like `compareWith`) to `validatePassword`.
If you don't like that, I still suggest to move `validatePassword2` to the shared location too. Something in utils maybe. And there we have all validation logic.

Reply:
Very right! Thanks for this good improvement idea.